SimG4Core/Application/interface/RunManagerMT.h

```diff
@@ -77,13 +77,17 @@ class RunManagerMT {
   // need a non-const pointer. Thread-safety is handled inside Geant4.
   inline PhysicsList* physicsListForWorker() const { return m_physicsList.get(); }
 
+  inline bool isPhase2() const { return m_isPhase2; }
```

Comment:
I think that these new methods/variables which include Phase2 in the name should be better renamed as (e.g.) `isHgcalMtd()` , `m_isHgcalMtd` , etc. because they do not refere to the Phase2 detector in general, but only to those two (three: HGCAL, ETL, BTL) subdetectors. Otherwise, named as they are now it can become misleading

Reply:
@perrotta , it is easy to implement, what you propose, but the choice if it is Phase2 or not in this PR means a choice of SteppingAction class, which is used for simulation of all sub-detectors. I would expect that in future some other parameters/classes will be chosen using this flag. Check on geometry does not introduce an uncertainty. For today I do not see any problem with it.

As an alternative, instead of checking on geometry this flag may be defined in configuration file g4SimHits by hand with modifiers....

Reply:
Ok, thank you @civanch 
Then I misunderstood: I thought that the choice of the SteppingAction was specific for every subdetector. If it is not so, then the name that you chose is correct